### PR TITLE
[Feat]: Authentication Implemented

### PR DESCRIPTION
**Pull request by parthpandyappp**
- Added `Login` functionality.
- Added `Logout` functionality.
- Added User Greetings on Login.

Note: Currently, the Auth only works for the user already created and saved in the `DB` and doesn't depend upon the inputs collected via form.

### CODE REVIEW

**Review by sandeepkumardev-zz — APPROVED** [review]

File: src/components/Nav.jsx [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+              src="https://img.icons8.com/ios/50/000000/retro-tv.png"
+              alt="logo"
+            />
+          </span>
+          /<span className="title-empha">Stre</span>Flix
+        </h2>
+      </Link>
+
+      <div className="nav-pills">
+        <div>{user ? <h4>Hello {user.firstName}👋</h4> : ""}</div>
```
Comment (sashaboi) [inline src/components/Nav.jsx:24]: very cool logic, simple and clean.

File: src/pages/Signup.jsx [hunk 2]
```diff
@@ -1,5 +1,47 @@
+import { Link } from "react-router-dom";
 function Signup() {
-  return <h1>Signup</h1>;
```
Comment (sashaboi) [inline src/pages/Signup.jsx:2]: Are you yet to do functionality here?

File: src/pages/Login.jsx [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+        <form
+          onSubmit={(e) => loginHandler(e, boolFunc, email, pass, navigate)}
+        >
+          <h2>Login</h2>
+          <div className="inp-unit">
+            <label for="email">Email</label>
+            <input
+              type="email"
+              name="email"
+              value="johnsangram@gkool.com"
```
Comment (sashaboi) [inline src/pages/Login.jsx:24]: is the login created just a placeholder to get over the hump and use apis? 
this value means user wont be able to log in with their own credentials?

File: src/helper-functions/auth-helpers.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+
+const loginHandler = async (e, boolFunc, email, pass, navigate) => {
+    e.preventDefault();
+    try {
+        const response = await axios.post(`/api/auth/login`, {
+            email: "gkool@neog.camp",
```
Comment (sashaboi) [inline src/helper-functions/auth-helpers.js:8]: it should mention in  your pr description that no matter what, only a default user is going to get logged in , in my opinion.

File: src/helper-functions/auth-helpers.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            password: "gkool",
+        });
+        boolFunc((prev) => !prev);
+
+        // saving the encodedToken in the localStorage
+        localStorage.setItem("token", response.data.encodedToken);
+        localStorage.setItem("userData", JSON.stringify(response.data.foundUser));
+        navigate("/");
+    } catch (error) {
+        console.log(error);
```
Comment (sashaboi) [inline src/helper-functions/auth-helpers.js:18]: could try to do error handling messages in the future with toasts or alerts.

File: src/pages/Home.jsx [hunk 6]
```diff
@@ -1,5 +1,89 @@
...
+          <Link to="/explore">
+            <button className="btn btn-primary">Explore Videos</button>
+          </Link>
+          <button className="btn btn-secondary ml-2">
+            <a href="#category">Stream by Categories</a>
+          </button>
+        </div>
+        <img
+          className="landing-img"
+          src="https://assets.website-files.com/5d5e2ff58f10c53dcffd8683/5d5e304c8f10c501d7fd8a1d_meditating.svg"
```
Comment (sashaboi) [inline src/pages/Home.jsx:33]: this is very cool , love the overall UI. do check contrast value to your primary color to white, only nitpicking doubt I found.

**Review by sashaboi — APPROVED** [review]
Hey man , I love the flow the hand login logout , and the Hey Parth greeting on top, seems very pro. UI is top notch, looks amazing.

I have made a few suggestions below, do go through them.
Also, the PR description should kind of mention that the user login is only going to be default mandatory and not going to be dependant on the form application anyway, misleads the reviwer. 
Onkar Deshpande

File: src/pages/Signup.jsx (hunk 2 above)
Comment (parthpandyappp) [reply src/pages/Signup.jsx:2]: Currently, Signup functionality is not in action.
Auth only works for the users already created in the db.

**Comment by parthpandyappp** [comment]
Hey Onkar, I loved the way you pumped me up with UI compliments :fire: 
Also, tremendous thanks for putting up your valuable feedback. I'll go ahead with merging meanwhile, I'll update the PR description so that even if anyone looks into the PR history might not get confused:+1:

**Comment by parthpandyappp** [comment]
Thankyou @sandeepkumardev for reviewing my PR :+1:

**Pull request merged**